runelite-client/src/main/java/net/runelite/client/plugins/gpu/GpuPlugin.java

```diff
@@ -243,6 +243,12 @@ protected void startUp()
 				modelBuffer = new GpuIntBuffer();
 
 				canvas = client.getCanvas();
+
+				if (!canvas.isDisplayable())
```

Comment:
You could just move this check to the very top to prevent allocating all of the buffers each tick